sdk/rust/src/consensus/zmq_driver.rs

```diff
@@ -200,7 +202,12 @@ pub fn register(
                     }
                     ConsensusRegisterResponse_Status::NOT_READY => {
                         thread::sleep(retry_delay);
-                        retry_delay = retry_delay * 2;
+                        if retry_delay < MAX_RETRY_DELAY {
```

Comment:
You could just use `retry_delay = ::std::cmp::min(retry_delay * 2, MAX_RETRY);` and avoid the big if block here